### PR TITLE
better recipe lookup

### PR DESCRIPTION
**Pull request by screret**
## What
remakes recipe lookup with specialized logic.

## Implementation Details
the Ingredient equality check might need some cleanup.

## Outcome
500-900% FASTER RECIPE LOOKUP

## Additional Information
https://media.discordapp.net/attachments/1089296351906504835/1203755074220720178/image.png?ex=65d23f46&is=65bfca46&hm=fb3c74269913573659c45161b453cc21c6fd09ae968226ad50fef9dd2a59603b&=&format=webp
https://media.discordapp.net/attachments/1089296351906504835/1203755074497806346/image.png?ex=65d23f46&is=65bfca46&hm=a702a7e825fb6049eb428b7043e02c5bf6c7f2f7880cbf212c8ac0bf8d6b0699&=&format=webp

## Potential Compatibility Issues
shouldn't cause any as long as nobody's changing the recipe logic internals too much, as this only touches the recipe lookup.

### CODE REVIEW

File: gradle/scripts/dependencies.gradle [hunk 1]
```diff
@@ -31,8 +31,8 @@ dependencies {
     modApi(include("com.tterrag.registrate:Registrate:${forge.versions.registrate.get()}"))
 
     // Create and its dependencies
-    modImplementation("com.simibubi.create:create-${libs.versions.minecraft.get()}:${forge.versions.createForge.get()}:all") { transitive = false }
-    modImplementation forge.flywheel
+    modCompileOnly("com.simibubi.create:create-${libs.versions.minecraft.get()}:${forge.versions.createForge.get()}:all") { transitive = false }
+    modCompileOnly forge.flywheel
```
Comment (mikerooni) [inline gradle/scripts/dependencies.gradle:35]: Is this intentional or accidentally committed?

**Review by mikerooni — APPROVED** [review]
lgtm

File: gradle/scripts/dependencies.gradle (hunk 1 above)
Comment (screret) [reply gradle/scripts/dependencies.gradle:35]: Oops lol

Comment (screret) [reply gradle/scripts/dependencies.gradle:35]: It kept crashing for me

File: src/main/java/com/gregtechceu/gtceu/api/recipe/ingredient/SizedIngredient.java [hunk 2]
```diff
@@ -119,6 +122,13 @@ public boolean isEmpty() {
         return inner.isEmpty();
     }
 
+    @Override
+    public int hashCode() {
+        int result = amount;
+        result = 31 * result + Arrays.hashCode(itemStacks);
```
Comment (PrototypeTrousers) [inline src/main/java/com/gregtechceu/gtceu/api/recipe/ingredient/SizedIngredient.java:128]: amount is discarded, is this intended?

**Pull request merged**